sdcm/sct_config.py

```diff
@@ -2289,6 +2293,66 @@ def verify_configuration_urls_validity(self):  # pylint: disable=too-many-branch
         get_branch_version_for_multiple_repositories(
             urls=(self.get(url) for url in repos_to_validate if self.get(url)))
 
+    def get_version_based_on_conf(self):
+        """
+        figure out which version and if it's enterprise version
+        base on configuration only, before nodes are up and running
+        so test configuration can set up things which need to happen
+        before nodes are up
+
+        this is information is cached on the SCTConfiguration object
+        :return: tuple - (scylla_version, is_enterprise)
+        """
+        backend = self.get('cluster_backend')
+        scylla_version = None
+        _is_enterprise = False
+
+        if not self.get('use_preinstalled_scylla'):
+            scylla_repo = self.get('scylla_repo')
+            scylla_version = get_branch_version(scylla_repo)
+            _is_enterprise = is_enterprise(scylla_version)
+        elif unified_package := self.get('unified_package'):
+            from sdcm.remote import LOCALRUNNER
+            import tempfile
+            from sdcm.remote import shell_script_cmd
```

Comment:
Why imports are here and not at the top of the file?

Reply:
fixed